Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -30,6 +30,27 @@ internal func withAutoReleasePool<T>(_ execute: () throws -> T) rethrows -> T {
 #endif
 }
 
+/// Information about an EventLoop tick
+public struct EventLoopTickInfo {
```

Comment:
- needs `NIO` prefix
- lacks identifier for EventLoop
- as said above, I think we should probably periodically give the user one value which contains information from _all_ loops in a group together. Something like


```
public struct NIOEventLoopMetrics {
    public struct LoopMetric {
        var loopID: ...
        var numberOfTasks: Int
        var startTime: NIODeadline
    }
    
    public var loopMetrics: [LoopMetric]
}
```

I know this raises a lot of questions but I think we should think through how a user would use this and then settle on what NIO should provide. Just calling the delegate _on each loop_ for every tick is easy but requires a lot of heavy-lifting on the user's part which seems bound to go wrong, no?

Reply:
IMO the trade-off falls the other way: we should offer the lowest level thing, which lets expert users get what they need, first. We can then do the engineering to produce something better. But frankly we've spent years not exposing these metrics because we wanted to do it right, and I'm a bit sick of stumbling around in the dark. So if we can get something good enough cheaply, we should.

Reply:
I agree with Cory here. Let's do the simplest approach and just expose a delegate that gets called every tick and users have to do the aggregation. Over time we will see what kind of aggregation users are doing and provide pre-backed delegates that give them an even simpler interface